### PR TITLE
Bucket Sort in Ruby

### LINKED ISSUE #2124: Bucket Sort
- [x] C
- [x] C++
- [ ] C# - @luv4bytes
- [ ] CoffeeScript
- [ ] Dart
- [x] Go - @m-code12
- [x] Java
- [x] JS - @abhu-A-J
- [x] Kotlin - @shubhammantri1
- [ ] Php - @srimani-programmer
- [x] Python
- [x] Readme
- [x] Ruby - @m-code12
- [ ] Rust
- [ ] TypeScript

### PR DESCRIPTION
**Pull request by m-code12**
Fixes #2124 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Bucket Sort in Ruby

#### Languages Used:
- Ruby

#### Files Added:
- Bucket_Sort.rb

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by m-code12** [comment]
I have implemented Bucket Sort in Ruby following all the required coding conventions. Please review and merge @jainaman224 @singh-shreya6

File: Bucket_Sort/Bucket_Sort.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+list = gets
+puts 'The sorted list is : '
+print bucket_sort(list)
+
+
+# Output :
+# Enter a list of numbers (separated by space) : 
+# 7 9 5 4 1
+# The sorted list is : 
+# 1 4 5 7 9
```
Comment (singh-shreya6) [inline Bucket_Sort/Bucket_Sort.rb:35]: Please add a blank line at end of file

File: Bucket_Sort/Bucket_Sort.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+    bucket_count = ((array.max - array.min) / bucket_size).floor + 1
+
+    # create buckets
+    buckets = []
+    bucket_count.times { buckets.push [] }
+
+    # fill buckets
+    array.each do |item|
+    buckets[((item - array.min) / bucket_size).floor].push(item)
```
Comment (singh-shreya6) [inline Bucket_Sort/Bucket_Sort.rb:16]: Please add 4 space indentation inside loop

File: Bucket_Sort/Bucket_Sort.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  end
```
Comment (singh-shreya6) [inline Bucket_Sort/Bucket_Sort.rb:17]: Please correct indentation of end

**Comment by m-code12** [comment]
I have made the changes as you have said .Please review and merge @singh-shreya6

**Pull request merged**